ESP_ATMod/command.cpp

```diff
@@ -2662,3 +2752,146 @@ void printCertificateName(uint8_t number)
 		Serial.println(F("cert ok"));
 	}
 }
+
+/*
+ * Compare wifi networks by RSSI
+ */
+int compWifiRssi(const void *elem1, const void *elem2)
+{
+	int f = *((int *)elem1);
+	int s = *((int *)elem2);
+	if (WiFi.RSSI(f) > WiFi.RSSI(s))
+		return -1;
+	if (WiFi.RSSI(f) < WiFi.RSSI(s))
+		return 1;
+	return 0;
+}
+
+/*
+ * Print scanned wifi network information
+ */
+void printCWLAP(int indices[], size_t size)
+{
+	for (size_t i = 0; i < size; i++)
+	{
+		bool show = false;
+		if (authmodeMask & (1 << WiFi.encryptionType(indices[i])))
+		{
+			show = true;
+		}
+		else if (WiFi.encryptionType(indices[i]) > 8)
+		{
+			show = true;
+		}
+
+		if (show)
+		{
+			if (WiFi.RSSI(indices[i]) > rssiFilter)
+			{
+				String result = "+CWLAP:(";
+
+				if (printMask & (1 << 0))
+				{
+					result += WiFi.encryptionType(indices[i]);
+					result += ",";
+				}
+				if (printMask & (1 << 1))
+				{
+					result += WiFi.SSID(indices[i]);
+					result += ",";
+				}
+				if (printMask & (1 << 2))
+				{
+					result += WiFi.RSSI(indices[i]);
+					result += ",";
+				}
+				if (printMask & (1 << 3))
+				{
+					result += WiFi.BSSIDstr(indices[i]).c_str();
+					result += ",";
+				}
+				if (printMask & (1 << 4))
+				{
+					result += WiFi.channel(indices[i]);
+					result += ",";
+				}
+				if (printMask & (1 << 5))
+				{
+					// freq_offset here
+					result += 0;
+					result += ",";
+				}
+				if (printMask & (1 << 6))
+				{
+					// freqcal_val here
+					result += 0;
+					result += ",";
+				}
+				if (printMask & (1 << 7))
+				{
+					// pairwise_cipher here
+					result += 0;
+					result += ",";
+				}
+				if (printMask & (1 << 8))
+				{
+					// group_cipher here
+					result += 0;
+					result += ",";
+				}
+				if (printMask & (1 << 9))
+				{
+					// bgn here
+					result += 0;
+					result += ",";
+				}
+				if (printMask & (1 << 10))
+				{
+					// wps here
+					result += 0;
+					result += ",";
+				}
+
+				// Remove trailing comma
+				result.remove(result.lastIndexOf(','));
+				result += ")";
+				Serial.printf("%s\n", result.c_str());
+			}
+		}
+	}
+
+	gsFlag_Busy = false;
+	Serial.printf_P(MSG_OK);
+}
+
+/*
+ * Print the networks found from scanning
+ */
+void prinScanResult(int networksFound)
+{
+	if (sort_enable == 0)
+	{
+		int indices[networksFound];
+		for (size_t i = 0; i < networksFound; i++)
+		{
+			indices[i] = i;
+		}
+
+		// Print the unsorted networks
+		printCWLAP(indices, sizeof indices / sizeof *indices);

```

Comment:
the size of indices array is `networkFound`, the expression `sizeof indices / sizeof *indices` seems to me to be correct by a coincidence only, because sizeof (*indices) == sizeof (int).

Reply:
That is true. I did this on purpose because I thought it is safer to actual check the length of the array instead of a number that should populate it. 

I suggest to keep it this way but I can change it to the same syntax you suggested in the next command:
`sizeof(indices) / sizeof(indices[0]), sizeof(indices[0])`

Reply:
I know this is the same but for clarity please change it. The reason is I have seen many time the notation I suggested but never the yours one. Although they are the same.

Reply:
You can see from my original comment, I was confused and initially thought the expression is wrong :)